core/src/main/java/ai/timefold/solver/core/impl/constructionheuristic/DefaultConstructionHeuristicPhase.java

```diff
@@ -24,17 +25,17 @@ public class DefaultConstructionHeuristicPhase<Solution_>
         implements ConstructionHeuristicPhase<Solution_> {
 
     protected final ConstructionHeuristicDecider<Solution_> decider;
-    protected final EntityPlacer<Solution_> entityPlacer;
+    protected final PlacerBasedMoveRepository<Solution_> moveRepository;
     private TerminationStatus terminationStatus = TerminationStatus.NOT_TERMINATED;
 
     protected DefaultConstructionHeuristicPhase(DefaultConstructionHeuristicPhaseBuilder<Solution_> builder) {
         super(builder);
         this.decider = builder.decider;
-        this.entityPlacer = builder.getEntityPlacer();
+        this.moveRepository = new PlacerBasedMoveRepository<>(builder.getEntityPlacer());
```

Comment:
Should we retrieve it from the builder?

Reply:
I'm only making minimal possible changes. Placers will eventually be going away anyway, as CH starts supporting move streams. I want to change the legacy code as little as possible.